airflow/sensors/base.py

```diff
@@ -72,6 +73,14 @@ def __bool__(self) -> bool:
         return self.is_done
 
 
+def coerce_timedelta(value: float | timedelta, *, key: str) -> timedelta:
+    if isinstance(value, timedelta):
+        return value
+    if isinstance(value, Number):
+        return timedelta(seconds=value)
+    raise ValueError(f"{key} should be timedelta object or a number of seconds")
```

Comment:
i think it's ok to fail here (though i'd add `if value is None: return None` at the top)

BUT, i think it would be best to help the user by failing fast i.e. check the type in operator init and raise accordingly (rather than waiting until code gets deployed and task runs)

Reply:
the other benefit is then... if we're failing fast... then we don't really need this `key` argument which, at call site, is a little confusing.

if you validate the param in `__init__` then you can immediately raise with a helpful message.